metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala

```diff
@@ -52,6 +52,27 @@ final class BuildTargets() {
   private val buildTargetInference =
     new ConcurrentLinkedQueue[AbsolutePath => Seq[BuildTargetIdentifier]]()
 
+  val buildTargetsOrder: BuildTargetIdentifier => Int = {
+    (t: BuildTargetIdentifier) =>
+      var score = 1
+
+      val isSupportedScalaVersion = scalaInfo(t).exists(t =>
+        ScalaVersions.isSupportedScalaVersion(t.getScalaVersion())
+      )
+      if (isSupportedScalaVersion) score <<= 2
+
+      val isJVM = scalacOptions(t).exists(_.isJVM)
+      if (isJVM) score <<= 1
+
+      // note(@tgodzik) once the support for Scala 3 is on par with Scala 2 this can be removed
+      val isScala2 = scalaInfo(t).exists(info =>
+        !ScalaVersions.isScala3Version(info.getScalaVersion())
+      )
+      if (isScala2) score <<= 1
+
+      score
+  }
```

Comment:
So I _think_ I understand what is going on here, but why when used up above do we so a `sortBy` and below we use `max`?. Also, what's the benefit of using binary shift operators here? I've actually never came across these in the wild before 😬

Reply:
We are using maxBy in a scenario that we know the list is not empty and it produces a single element. `sortBy` still produces a list, which is useful in the other scenario.

Binary shift is I think just an easy way to introduce priority.

Reply:
Sounds good. Thanks for explaining.